tms/src/main/java/com/netcracker/edu/tms/service/MailServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package com.netcracker.edu.tms.service;
+
+import com.netcracker.edu.tms.model.Mail;
+import org.springframework.scheduling.concurrent.CustomizableThreadFactory;
+import org.springframework.stereotype.Service;
+
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+@Service
+public class MailServiceImpl implements MailService {
+
+    private final String usernameForSession = "dmitrybobryakov@gmail.com";
+    private final String passwordForSession = "#Steam373";
+    private ExecutorService executor = Executors.newFixedThreadPool(10,
+            new CustomizableThreadFactory("MailSender executor- "));
+
+    @Override
+    public void send(String address, Mail mail) {
+        Session currentSession = this.setPropertiesAndGetSession();
+        MailSender sender = new MailSender(currentSession, usernameForSession, address, mail);
+        executor.execute(sender);
+    }
+
+    @Override
+    public void send(List<String> addresses, Mail mail) {
+        Session currentSession = this.setPropertiesAndGetSession();
+        for (String oneAddress : addresses) {
+            MailSender sender = new MailSender(currentSession, usernameForSession, oneAddress, mail);
+            executor.execute(sender);
+        }
+    }
+
+    private Session setPropertiesAndGetSession() {
```

Comment:
each method invocation creates new session
let's store session in field
and reuse it while emailing